examples/hb-subset.example.node.js

```diff
@@ -23,12 +21,19 @@ const SUBSET_TEXT = 'abc';
 
     /* Add your glyph indices here and subset */
     const input = exports.hb_subset_input_create_or_fail();
-    const unicode_set = exports.hb_subset_input_unicode_set(input);
-    for (const text of SUBSET_TEXT) {
-        exports.hb_set_add(unicode_set, text.codePointAt(0));
+    const plan = exports.hb_subset_plan_create_or_fail(face, input);
+    console.log('plan: ', plan)
+    const glyph_map = exports.hb_subset_plan_old_to_new_glyph_mapping(plan);
+    console.log('glyph_map: ', glyph_map)
```

Comment:
@garretrieger I'm sorry to say that I'm a rookie when it comes to `harfbuzz`. Can you guide how to accomplish something like `subset-gf-icons`?

Reply:
So what you want to do is hits:
1. create a subset input object.
2. Populate it with the gids or codepoints for the icon glyphs you want to keep (but not gids/codepoints for a-z/A-Z). That's done by adding values to hb_subset_input_glyph_set or hb_subset_input_unicodes_set.
3. Create a subset plan via hb_subset_plan_create_or_fail with the input from 1. as an argument.
4. Get the old to new glyph mapping from hb_subset_plan_old_to_new_glyph_mapping
5. Iterate through the keys in this map. These are gids which are needed in the subset. Add them to the input from 1.
6. Set the no layout closure flag on the input from 1.
7. Add remaining glyphs needed from a-z/A-Z t to the input from 1.
8. Actually run the subset via hb_subset_or_fail

Reply:
Is the [code here ](https://github.com/adobe/binned-ift-reference/blob/2629ce830c09ccd239ce2647bf05c0989bfd8bb8/src/randtest.cc#L104)available for reference?

I found that `hb_map_keys` is used here in step 5, but currently harfbuzzjs doesn't seem to export this function.

Reply:
Which function is used in the first step?

Reply:
Thank you for your guidance. I have changed the logic in JS, can you please review it?

Reply:
The only difference now is that GlyphID: `star` retains the Glyph name, while subset-gf-icons(see .ttx file) does not.

@rsheeter Can you confirm that this will be a problem?

```xml
 <GlyphID id="0" name=".notdef"/>
 <GlyphID id="1" name="uniE838"/>
 <GlyphID id="2" name="glyph00002"/>
 <GlyphID id="3" name="a"/>
 <GlyphID id="4" name="r"/>
 <GlyphID id="5" name="s"/>
 <GlyphID id="6" name="t"/>
```

Reply:
The glyph names shouldn't matter as long as the right glyphs and rules are being retained.

Reply:
@rsheeter Thanks.

There are a couple of other places that are different and I'm not sure which is correct.

![image](https://github.com/yisibl/harfbuzzjs/assets/2784308/21e1e91e-0d69-476d-9579-443e9be11d1c)
![image](https://github.com/yisibl/harfbuzzjs/assets/2784308/ea17d6b0-70c8-4b9f-9d2f-28667a4d6b77)

Reply:
Those seem surprisingly different if it's retaining the same glyphs with the same outlies. @garretrieger is this expected for some reason?

Reply:
For 2: So, does Harfbuzz not recalculating `maxp` have any effect on the font display?

> Ok for 1. for the a-z glyphs you add those via codepoint value, so you only get the lower case version of them retained. This is expected. If you want to get the upper and lower case you'll need to manually add the uppercase version of each.

Yes, I added uppercase and now the `cmap` table matches the results of subset-gf-icons.